python_modules/dagster/dagster/core/execution_plan/multiprocessing_engine.py

```diff
@@ -11,57 +10,47 @@
 
 from .create import create_execution_plan_core
 from .intermediates_manager import FileSystemIntermediateManager
-from .objects import (
-    ExecutionPlan,
-    ExecutionStepEvent,
-    StepFailureData,
-    StepOutputHandle,
-    StepOutputData,
-    ExecutionStepEventType,
-)
+from .objects import ExecutionPlan, ExecutionStepEvent
 from .simple_engine import start_inprocess_executor
 
 
-def _execute_in_child_process(queue, environment_dict, run_config, step_key, input_meta_dict):
+CHILD_PROCESS_DONE_SENTINEL = 'CHILD_PROCESS_DONE_SENTINEL'
+CHILD_PROCESS_SYSTEM_ERROR_SENTINEL = 'CHILD_PROCESS_SYSTEM_ERROR_SENTINEL'
+
+
+def _execute_in_child_process(queue, environment_dict, run_config, step_key):
     from dagster.core.execution import yield_pipeline_execution_context
 
     check.inst(run_config.executor_config, MultiprocessExecutorConfig)
-    check.dict_param(input_meta_dict, 'input_meta_data', key_type=str, value_type=StepOutputHandle)
 
     pipeline = run_config.executor_config.pipeline_fn()
 
-    with yield_pipeline_execution_context(
-        pipeline, environment_dict, run_config.with_tags(pid=str(os.getpid()))
-    ) as pipeline_context:
+    try:
 
-        intermediates_manager = FileSystemIntermediateManager(pipeline_context.files)
+        with yield_pipeline_execution_context(
+            pipeline, environment_dict, run_config.with_tags(pid=str(os.getpid()))
+        ) as pipeline_context:
 
-        execution_plan = create_execution_plan_core(pipeline_context)
+            intermediates_manager = FileSystemIntermediateManager(pipeline_context.files)
 
-        for step_event in start_inprocess_executor(
-            pipeline_context, execution_plan, intermediates_manager, step_keys_to_execute=[step_key]
-        ):
-            data_to_put = {
-                'event_type': step_event.event_type,
-                'step_output_data': {
-                    'step_output_handle': step_event.step_output_data.step_output_handle,
-                    'value_repr': step_event.step_output_data.value_repr,
-                }
-                if step_event.event_type == ExecutionStepEventType.STEP_OUTPUT
-                else None,
-                # TODO actually marshal data
-                'step_failure_data': None,
-                'step_key': step_event.step.key,
-                'tags': step_event.tags,
-            }
+            execution_plan = create_execution_plan_core(pipeline_context)
 
-            queue.put(data_to_put)
+            for step_event in start_inprocess_executor(
+                pipeline_context,
+                execution_plan,
+                intermediates_manager,
+                step_keys_to_execute=[step_key],
+            ):
+                queue.put(step_event)
 
-    queue.put('DONE')
-    queue.close()
+        queue.put(CHILD_PROCESS_DONE_SENTINEL)
+    except:  # pylint: disable=bare-except
+        queue.put(CHILD_PROCESS_SYSTEM_ERROR_SENTINEL)
```

Comment:
I'd really like to have test coverage for this line

Reply:
so as currently structured this is very difficult to test, as you need to inject a *system* error. we can work on restructuring this in a subsequent diff

Reply:
https://github.com/dagster-io/dagster/pull/893 to address this